test/e2e/test_cap_kill.py

```diff
@@ -140,3 +142,177 @@ def test_cap_kill_not_inherited_by_running_jobs(
+                            },
+                        },
+                        "embeddedFiles": [
+                            {
+                                "name": "CheckCaps",
+                                "type": "TEXT",
+                                "runnable": True,
+                                "data": "capsh --print",
+                            },
+                        ],
+                    },
+                },
+            ],
+        },
+    )
+    job.wait_until_complete(client=deadline_client)
+
+    # THEN
+    #
+    # The output of `capsh --print` will look something like this:
+    # ````
+    # Current: =
+    # Bounding set =cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read,cap_perfmon,cap_bpf,cap_checkpoint_restore
+    # Ambient set =
+    # Current IAB:
+    # Securebits: 00/0x0/1'b0 (no-new-privs=0)
+    #  secure-noroot: no (unlocked)
+    #  secure-no-suid-fixup: no (unlocked)
+    #  secure-keep-caps: no (unlocked)
+    #  secure-no-ambient-raise: no (unlocked)
+    # uid=1001(ssm-user) euid=1001(ssm-user)
+    # gid=1001(ssm-user)
+    # groups=1001(ssm-user)
+    # Guessed mode: UNCERTAIN (0)
+    # ````
+    # The information we care about is `Current` and `Ambient`. We want these to be empty for the current process (meaning the job process)
+    # because this means that the process has no capabilities. We don't care about `Bounding set` as it is not capabilities the process actually has.
+    job.refresh_job_info(client=deadline_client)
+    assert job.task_run_status == TaskStatus.SUCCEEDED
+    logs_client = boto3.client(
+        "logs",
+        config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+    )
+    job.assert_single_task_log_contains(
+        deadline_client=deadline_client,
+        logs_client=logs_client,
+        expected_pattern=re.escape("Current: =\n"),
+    )
+
+    job.assert_single_task_log_contains(
+        deadline_client=deadline_client,
+        logs_client=logs_client,
+        expected_pattern=re.escape("Ambient set =\n"),
+    )
+
+
+@pytest.mark.skipif(
+    os.environ["OPERATING_SYSTEM"] == "windows",
+    reason="Linux specific test",
+)
+@pytest.mark.usefixtures("session_worker")
+def test_worker_only_has_cap_kill(
+    deadline_client: DeadlineClient,
+    deadline_resources: DeadlineResources,
+) -> None:
+    """Tests to make sure that the worker agent has no capabilities besides cap_kill"""
```

Comment:
Can we mention that this tests that the worker agent has `CAP_KILL` in its effective and permitted capability sets but NOT its inheritable capability set?